python/pyspark/rdd.py

```diff
@@ -141,7 +141,9 @@ def _parse_memory(s):
 
 
 def _load_from_socket(sock_info, serializer):
-    (sockfile, sock) = local_connect_and_auth(*sock_info)
+    port = sock_info[0]
+    auth_secret = sock_info[1]
+    (sockfile, sock) = local_connect_and_auth(port, auth_secret)
```

Comment:
This was correctly ported back at https://github.com/apache/spark/pull/25593. There's no such code you pointed out in Spark 2.4.4: https://github.com/apache/spark/blob/v2.4.4/python/pyspark/sql/dataframe.py#L2182.

Are you using your own fork or mixing the Spark versions? Your error message seems from https://github.com/apache/spark/blob/v2.4.3/python/pyspark/sql/dataframe.py#L2179 which is Spark 2.4.3. Spark 2.4.3 does not have this change.

Reply:
@HyukjinKwon You are linking to `dataframe.py` but I patched `rdd.py`
The code is there:
https://github.com/apache/spark/blob/v2.4.4/python/pyspark/rdd.py#L144

So `(*sock_info)` is used instead of passing individual touple values.

Reply:
Ok I see now. It is possible that that I messed something while using pipenv. 
Anyway I feel that using `(*sock_info)` is inherently unsafe and prone to such errors.

Reply:
Yeah, I actually already pointed it out it's error-prone. Feel free to open another PR to fix it.